apollo-client/src/main/java/com/ctrip/framework/apollo/util/factory/DefaultPropertiesFactory.java

```diff
@@ -0,0 +1,42 @@
+package com.ctrip.framework.apollo.util.factory;
+
+import com.ctrip.framework.apollo.util.OrderedProperties;
+import com.google.common.base.Strings;
+import java.util.Properties;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Default PropertiesFactory implementation.
+ * @author songdragon@zts.io
+ */
+public class DefaultPropertiesFactory implements PropertiesFactory {
+
+  private static final Logger logger= LoggerFactory.getLogger(DefaultPropertiesFactory.class);
+
+  protected boolean isOrderEnabled=false;
+
+  public DefaultPropertiesFactory(){
+    initOrderEnabled();
+  }
+
+  private void initOrderEnabled() {
+    String customizedOrderEnable = System.getProperty(APOLLO_PROPERTY_ORDER_ENABLE);
```

Comment:
I think we could support setting this flag in more ways. 

e.g. 

1. app.properties
```java
Foundation.app().getProperty(APOLLO_PROPERTY_ORDER_ENABLE, null);
```

2. Spring environment
Just add it to `com.ctrip.framework.apollo.spring.boot.ApolloApplicationContextInitializer#APOLLO_SYSTEM_PROPERTIES`

https://github.com/ctripcorp/apollo/blob/42edb9da963df986538edcda4ff58d474fb3abed/apollo-client/src/main/java/com/ctrip/framework/apollo/spring/boot/ApolloApplicationContextInitializer.java#L63